### PR TITLE
Initialize the foreground and background in TermControl's constructor

### PR DESCRIPTION
**Pull request by PankajBhojwani**
We recently moved setting the foreground and background out of
`TermControl::_ApplyUISettings` , but that means we no longer set the
foreground/background in the constructor, resulting in the terminal
being gray on startup until we call `UpdateAppearance` after the
renderer is setup. This change sets the foreground and background in the
TermControl constructor to avoid this. 

## Validation Steps Performed
Terminal no longer starts up gray

### CODE REVIEW

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -194,6 +194,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         _autoScrollTimer.Tick({ this, &TermControl::_UpdateAutoScroll });
 
         _ApplyUISettings(_settings);
+
+        // Set the foreground and background here in the constructor
+        // (they will also be called later in UpdateAppearance, but we want to have it here so
+        //  that the terminal is not gray on startup)
+        const auto bg = _settings.DefaultBackground();
+        _BackgroundColorChanged(bg);
```
Comment (DHowett) [inline src/cascadia/TerminalControl/TermControl.cpp:202]: I'm actually.. afraid of this. BackgroundColorChanged _is also a fire_and_forget_ which we should not use from the constructor (!)

Comment (PankajBhojwani) [reply src/cascadia/TerminalControl/TermControl.cpp]: Ah, thanks for pointing that out. Fixed it here and in `_UpdateAppearanceFromUIThreadUnderLock`

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 2]
```diff
@@ -198,8 +198,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         // Set the foreground and background here in the constructor
         // (they will also be called later in UpdateAppearance, but we want to have it here so
         //  that the terminal is not gray on startup)
-        const auto bg = _settings.DefaultBackground();
-        _BackgroundColorChanged(bg);
+        til::color newBgColor{ _settings.DefaultBackground() };
```
Comment (DHowett) [inline src/cascadia/TerminalControl/TermControl.cpp:201]: i suspect you know what i am going to say about us having three copies of this code ;P

Comment (PankajBhojwani) [reply src/cascadia/TerminalControl/TermControl.cpp:201]: ahh haha alright that's fair... I was just trying to have it so there's only 1 update appearance call but okay will make a helper for the fg/bg

**Comment by DHowett** [comment]
We're going to ship w/the gray flash. I'll keep this marked for Servicing.

**Review by zadjii-msft — APPROVED** [review]

**Comment by DHowett** [comment]
Wow, this is still relevant @zadjii-msft with your split?

**Comment by PankajBhojwani** [comment]
I cannot repro the 'starting up gray' issue so I'm closing this!

**Pull request closed without merging**